File: lib_arch/src/main/java/com/yyxnb/arch/base/IFragment.java
package com.yyxnb.arch.base;

import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.OnLifecycleEvent;
import android.os.Bundle;

import com.yyxnb.arch.delegate.FragmentDelegate;
import com.yyxnb.arch.utils.AppManager;

public interface IFragment extends IView {

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    default FragmentDelegate getBaseDelegate() {
        return AppManager.getInstance().getFragmentDelegate(this, hashCode());
    }

    /**
     * 用户可见时候调用
     */
    default void onVisible() {
    }

    /**
     * 用户不可见时候调用
     */
    default void onInVisible() {
    }

    default Bundle initArguments() {
        return null;
    }

    default String sceneId() {
        return null;
    }

}

File: lib_arch/src/main/java/com/yyxnb/arch/base/IFragment.kt
package com.yyxnb.arch.base

import android.os.Bundle
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.OnLifecycleEvent
import com.yyxnb.arch.delegate.FragmentDelegate
import com.yyxnb.arch.utils.AppManager

/**
 * Fragment 需实现
 *
 * @author yyx
 */
interface IFragment : IView {

    /**
     * Fragment 代理
     *
     * @return FragmentDelegate
     */
    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun getBaseDelegate(): FragmentDelegate = AppManager.getFragmentDelegate(this, hashCode())

    /**
     * 用户可见时候调用
     */
    fun onVisible() {}

    /**
     * 用户不可见时候调用
     */
    fun onInVisible() {}

    /**
     * 传值
     *
     * @return Bundle
     */
    fun initArguments(): Bundle? = null

    /**
     * id
     *
     * @return id
     */
    fun sceneId(): String? = null
}